Assets/Plugins/Source/EOS_SDK/Core/ConvertHelper.cs

```diff
@@ -71,7 +71,9 @@ private static void Convert(string from, out byte[] to, int fromLength)
 				from = "";
 			}
 
-			to = Encoding.ASCII.GetBytes(new string(from.Take(fromLength).ToArray()).PadRight(fromLength, '\0'));
+			to = new byte[fromLength];
+			ASCIIEncoding.ASCII.GetBytes(from, 0, from.Length, to, 0);
+			to[from.Length] = 0;  //  Null terminator at the end'\0'
```

Comment:
I'm somewhat surprised reading this.

We have a length `fromLength`. We make a byte array that is of size `fromLength`. Then, we set a variable at the index of `from.Length`. `from` is a string. If `from` has 10 characters in it, and `fromLength` is 10, this'd result in trying to put a 0 at index 10, which would be an out of bounds array.

Is this somehow set up so that every provided `from` is smaller than `fromLength`?

This looks like it would be different from `PadRight`, which would (I guess) put 0 in all empty array indexes to the right.